From: Juergen Beisert <jbe@pengutronix.de>
Date: Fri, 15 Nov 2013 14:40:49 +0100
Subject: [PATCH] ARM/AM335x: provide a CPU frequency driver Author: Markus
 Pargmann <mpa@pengutronix.de>

Enable the usage of the cpufreq-cpu0 driver to handle the core CPU
frequencies. Everything else is prepared, only the trigger for this
driver is missing yet.
This patch is required until device tree is supported for this feature.

Signed-off-by: Markus Pargmann <mpa@pengutronix.de>
Signed-off-by: Juergen Beisert <jbe@pengutronix.de>
---
 arch/arm/mach-omap2/board-generic.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/mach-omap2/board-generic.c b/arch/arm/mach-omap2/board-generic.c
index 87162e1..0890291 100644
--- a/arch/arm/mach-omap2/board-generic.c
+++ b/arch/arm/mach-omap2/board-generic.c
@@ -53,6 +53,8 @@ static void __init legacy_init_ehci_clk(char *clkname)
 
 static void __init omap_generic_init(void)
 {
+	struct platform_device_info devinfo = { .name = "cpufreq-cpu0", };
+
 	omap_sdrc_init(NULL, NULL);
 
 	of_platform_populate(NULL, omap_dt_match_table, NULL, NULL);
@@ -70,6 +72,8 @@ static void __init omap_generic_init(void)
 		omap_4430sdp_display_init_of();
 	else if (of_machine_is_compatible("ti,omap5-uevm"))
 		legacy_init_ehci_clk("auxclk1_ck");
+
+	platform_device_register_full(&devinfo);
 }
 
 #ifdef CONFIG_SOC_OMAP2420
